### ISSUE TITLE
Tutorial on XSL

### ISSUE THREAD

**Issue by arojascastro** [programminghistorian/jekyll#794]
I am continuing translating the tutorial on xslt and found out that the numbering of the figures is wrong in the [English original](https://programminghistorian.org/lessons/transforming-xml-with-xsl#creating-and-testing-your-xsl-file). There is a Figure 3 and a Figure 5 but no Figure 4. It is a little error, but it affects Figures 5, 6 and 7. 

I can change it myself if you want, but I guess I should ask the editor @acrymble

**Comment by arojascastro** [comment]
Also for the record, I found a typo here: `the following exercies are located`

**Comment by arojascastro** [comment]
In this [solution](https://programminghistorian.org/lessons/transforming-xml-with-xsl#introduction-primary-source-reader): `<xsl:if test="keywords/keyword = 'slave rebellion'">` has to be `<xsl:if test="keywords/keyword = 'slave insurrections'">`. Otherwise, the output is empty (unless that is intentional, which would be of little help).

**Comment by acrymble** [comment]
Thanks @arojascastro I will make a pull request for the first 2. @mhbeals can you verify the above comment about slave rebellion? You know xsl much better than me.

**Comment by mhbeals** [comment]
Thanks @arojascastro.  Yes, that is an artifact of a previous version of the XML which had both but I seemed to have merged them for the simplified version of the XML.  Please change to insurrections.

**Comment by arojascastro** [comment]
Thanks. My translation is already corrected.

However, I'm having some problems with using the browser to transform the XML. While translating I was using Oxygen and I did not encounter any problem. After finishing the translation, I tried to reproduce the images and thus to use the browser transformation to take snapshoots. 

Here are the problems:

- I am using a Mac and Internet Explorer is not longer available for this OS.
- the new Microsoft browser (Microsoft Edge) does not support XSLT any longer - as far as I could see using Google
- so I tried Firefox and Chrome and I could not get the desired output (but it works with Oxygen). 

Can @mhbeals try it again with the latest version of all those browsers maybe?

If I am not mistaken, I think we have a sustainability problem here. I would suggest to rewrite this [section](https://programminghistorian.org/lessons/transforming-xml-with-xsl#creating-and-testing-your-xsl-file) because it assumes that we can use Internet Explorer (a very old version though). 

Possible solutions:
1. to use Oxygen (it is not free, but people can get temporary licenses and it's the de facto editor for this tasks. I work with Oxygen every day).
2. to implement one of the solutions explained in [W3schools](https://www.w3schools.com/xml/xsl_server.asp), but it seems too advanced for me.
3. to use the online editor [XSL transform](http://xsltransform.net/).

I would avoid number 2, explain number 1 and try to implement number 3. However, this editor may not be available in the future (it has ups and downs). 

This may seem a lot of work, but it isn't. The tutorial works fine except the section that I linked.

**Comment by arojascastro** [comment]
A colleague of mine (Peter Dängeli) suggests one more solution:

4. to use [Saxon CE](http://www.saxonica.com/ce/index.xml) - but this needs further comments about the installation process. 

On the other hand, he says that maybe the browser transformation with Chrome and Firefox is not working because of safety seetings that I need to change - like `--allow-file-access-from-files` [LINK](https://stackoverflow.com/questions/18586921/how-to-launch-html-using-chrome-at-allow-file-access-from-files-mode)

Is that possible @mhbeals ?

**Comment by mhbeals** [comment]
I must admit @arojascastro I am slightly perplexed.  I have tried the tutorial again and all the examples seem to work fine in Firefox & Edge, albeit the Windows versions. I do not have access to a MacOS system at the moment (I am abroad working from a Win10 laptop) so I cannot troubleshoot effectively.

In regard to your suggestions
1) I would rather not suggest Oxygen, owing as much to performance issues on some machines as the implicit endorsement of a costly product
2) I agree this might be beyond the remit of this lesson
3) This could work, but sort of defeats the purpose of low-cost, offline transformation
4) This is the subject of the part 2 tutorial I am writing now, so probably best not to overload this one

This leaves me with the idea that maybe its just the security settings on your system. Chrome will almost never work without extreme fiddling but Firefox should out of the box.  Can you attach your xml/xsl directly so I can see them? If it works on Oxygen they should be fine but just in case.

**Comment by mhbeals** [comment]
@arojascastro I think I may have found the problem. Are you referring specifically to the introduction example? I have found that this no longer works because of the ```<html> <body> </body>``` and ```</html>``` tags. Simply remove these (and fix the 'slave insurrections' tag) and it should hopefully render

**Comment by acrymble** [comment]
Just so I'm clear, we have 3 outstanding problems?:

- [ ] <xsl:if test="keywords/keyword = 'slave rebellion'"> should be <xsl:if test="keywords/keyword = 'slave insurrections'">
- [ ] https://programminghistorian.org/lessons/transforming-xml-with-xsl#creating-and-testing-your-xsl-file may not work on OS X?
- [ ] the 'introduction example' no longer works?

**Comment by arojascastro** [comment]
Hi there, 

I tested again, and fortunately it is working with Firefox in my MacOS. I failed with Chrome (maybe I need to change the safety settings) and I couldn't try with Internet Explorer (because I do not have it and I cannot install it). But I guess the problem is solved now. Apologies for my previous message. 

So @acrymble  I would say we only need to change this: 

<xsl:if test="keywords/keyword = 'slave rebellion'"> should be <xsl:if test="keywords/keyword = 'slave insurrections'">

Thanks.
